teps/0144-param-enum.md

```diff
@@ -212,8 +214,7 @@ While we haven't identified any `enum` use case in this scenario at the time wri
 ### Pipelines
 `Pipeline` author can also define a `string` `param` with the `enum` keyword in the `PipelineSpec` `Param` and pass this `string` `param` to the referenced `Task`.
 
-> :warning: Tekton validates the **intersection** of enum set specified in a PipelineTask and the enum set specified in the Pipeline (Params without
- enums specified refer to implicit enum sets with all possible values).
+> :warning: If both the `Pipeline` and `PipelineTasks`(embedded or referenced) specify an enum, the `enum` in the `Pipeline` must be a **subset** of the corresponding `enum` in the `PipelineTask`.
```

Comment:
I would like in future to be able to have a pipeline parameter where some of the values are handled by one Task and others by a different task. 

Example:
- a pipeline to build an artifact for multiple architectures and OSes
- one task can build linux artifact on multiple archs, its param enum is restricted to linux 
- another task can build windows artifact on multiple archs,  its param enum is restricted to windows
- the pipeline could accept a list of os/arch for which the artifact needs to be built, and its param enum could accept all linux and windows archs

I initially thought it would be possible to implement this, but actually, I'm not sure we can combine "when" expressions with matrix, so it would be difficult to split the input pipeline parameter into two arrays, one for each of the build task types.

So for now it's ok to only support subset, but I think there may be use cases in future where we might want to extend to intersection, as it was originally designed.

Reply:
Thanks for the input Andrea. We can revisit the scenario if we need it in the future, happy to help!